engine/src/main/java/org/hibernate/validator/internal/engine/validationcontext/AbstractValidationContext.java

```diff
@@ -434,6 +450,10 @@ private static final class BeanPathMetaConstraintProcessedUnit {
 
 		@Override
 		public boolean equals(Object o) {
+			if ( o == null ) {
```

Comment:
Since these are private classes for internal use only, we can ignore the null check, since we control where we are adding instances of these classes to `Set`s

Reply:
ok.